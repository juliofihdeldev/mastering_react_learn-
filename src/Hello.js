import React from "react";

export default function Hello({ name }) {
  if (name) {
    return <h1>Hello, {name}!</h1>;
  } else {
    return <span>Hey, stranger</span>;
  }
}