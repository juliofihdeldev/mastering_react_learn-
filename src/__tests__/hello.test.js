import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Hello from '../Hello';



test('Link change la classe lorsqu\'on le survole', () => {
  const component = renderer.create(
    <Hello name="http://www.facebook.com"/>
  );
    act(() => {
        /* fire events that update state */
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});