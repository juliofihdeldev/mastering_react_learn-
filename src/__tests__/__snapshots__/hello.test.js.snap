// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link change la classe lorsqu'on le survole 1`] = `
<h1>
  Hello, 
  http://www.facebook.com
  !
</h1>
`;
